### ISSUE TITLE
[CS] Hud and Radar Request

### ISSUE THREAD

**Issue by ChunkyLover53** [ValveSoftware/halflife#1959]
Allow custom color and transparency levels for Heads Up Display and Radar

**Comment by SpaceArbalest** [comment]
I personally believe that the radar in Counter-Strike 1.6 should at least be changed to a mini-map radar like the one in CS: Source. This would prevent getting lost in unfamiliar maps and allow players to know the exact positions of their teammates. Otherwise, players are likely to get lost and unable to accurately determine their teammates' positions, which would completely ruin the gaming experience.
As shown in the picture

[](url)

![Image](https://github.com/user-attachments/assets/7702b8fc-8bf8-42cb-9737-8fcafec287b7)

**Comment by di57inct** [comment]
what mod are you using for the radar?

**Comment by SpaceArbalest** [comment]
> what mod are you using for the radar?

Hello, bro. This image is just a reference schematic. However, I need to tell you that there are indeed people who have reverse-engineered CS1.6 to make it more refined, such as the mini-map radar I mentioned earlier. If you want to play CS1.6 with the mini-map radar I referred to before, but haven't yet obtained the mini-map radar implemented in code, you can check gamebanana.com to see if there are any related plugins available.

**Comment by di57inct** [comment]
> Hello, bro. This image is just a reference schematic. However, I need to tell you that there are indeed people who have reverse-engineered CS1.6 to make it more refined, such as the mini-map radar I mentioned earlier. If you want to play CS1.6 with the mini-map radar I referred to before, but haven't yet obtained the mini-map radar implemented in code, you can check gamebanana.com to see if there are any related plugins available.

there arent any working ones afaik, but there is
 https://github.com/hzqst/MetaHookSv and it would just need a working plugin for the radar.

**Comment by SpaceArbalest** [comment]
> > Hello, bro. This image is just a reference schematic. However, I need to tell you that there are indeed people who have reverse-engineered CS1.6 to make it more refined, such as the mini-map radar I mentioned earlier. If you want to play CS1.6 with the mini-map radar I referred to before, but haven't yet obtained the mini-map radar implemented in code, you can check gamebanana.com to see if there are any related plugins available.
> 
> there arent any working ones afaik, but there is https://github.com/hzqst/MetaHookSv and it would just need a working plugin for the radar.

Please wait a moment; I am trying to search for available CS1.6 mini-map radar plugins using other languages. Additionally, the developer I mentioned earlier, who is working on decompiling CS1.6, I will notify you immediately once they have replaced the radar in defuse mode, rescue mode (hostage), and assassination VIP mode. Currently, this developer has only developed Team Deathmatch Mode, Zombie Mode, and Zombie Escape Mode (implemented via code) and has added a mini-map radar in Team Deathmatch mode, Zombie Mode, and Zombie Escape Mode.

**Comment by di57inct** [comment]
i have already looked and could only find an old one that isn't compatible with the new metahook. having the whole client dll modified is not viable in my opinion as long as there are modular options like metahooksv

**Comment by SpaceArbalest** [comment]
> i have already looked and could only find an old one that isn't compatible with the new metahook. having the whole client dll modified is not viable in my opinion as long as there are modular options like metahooksv

Modifying the game code allows for direct access to high-performance C++ code and resources, reducing the overhead of intermediate layers, interpretation, and conversion, thereby improving runtime efficiency.

**Comment by SpaceArbalest** [comment]
> i have already looked and could only find an old one that isn't compatible with the new metahook. having the whole client dll modified is not viable in my opinion as long as there are modular options like metahooksv

Yes, I encountered the exact same situation. I searched through all the major mainstream websites as well as Counter-Strike plugin download sites, such as YouTube, Bilibili (the Chinese version of YouTube), TikTok, Douyin (the Chinese version of TikTok), and Reddit. I just couldn't find a minimap radar plugin that's compatible with the latest Steam version of CS 1.6 and the plugin framework.
I'm really sorry about that. However, as I mentioned earlier, once a minimap radar implemented through code is developed, I will notify you at the earliest opportunity.

**Comment by SpaceArbalest** [comment]
> i have already looked and could only find an old one that isn't compatible with the new metahook. having the whole client dll modified is not viable in my opinion as long as there are modular options like metahooksv

Yes, I encountered the exact same situation. I searched through all the major mainstream websites as well as Counter-Strike plugin download sites, such as YouTube, Bilibili (the Chinese version of YouTube), TikTok, Douyin (the Chinese version of TikTok), and Reddit. I just couldn't find a minimap radar plugin that's compatible with the latest Steam version of CS 1.6 and the plugin framework.
I'm really sorry about that. However, as I mentioned earlier, once a minimap radar implemented through code is developed, I will notify you at the earliest opportunity.

**Comment by di57inct** [comment]
thank you, but I can't use it as im already using metahooksv and it needs a clean default client dll to read the signatures from. it has a lot of useful features that i got used to and can't play without anymore. here's a short showcase: https://youtube.com/shorts/83C-e_xrQ1g

**Comment by SpaceArbalest** [comment]
> thank you, but I can't use it as im already using metahooksv and it needs a clean default client dll to read the signatures from. it has a lot of useful features that i got used to and can't play without anymore. here's a short showcase: https://youtube.com/shorts/83C-e_xrQ1g

I just watched many of your videos on YouTube, which is why I'm only replying now. I’m very glad to be of help! Additionally, if the developer I mentioned—the one who reverse-engineered CS 1.6—also provides an interface for adding and modifying in-game items, along with a plugin entry point for the client, then you'll be able to play the modded version of CS 1.6 I referred to, which was further developed by this developer.

**Comment by di57inct** [comment]
alright, thanks :)

**Comment by SpaceArbalest** [comment]
> alright, thanks :)

You are welcome!
